### DISCUSSION TITLE
Add custom assertion for type already handled by FA

### DISCUSSION THREAD

**Discussion by lonix1** [fluentassertions/fluentassertions#1834] [Q&A] [answered]
I found this sample that creates a custom assertion for `MyClass`:
````cs
public class MyClassAssertions : ReferenceTypeAssertions<MyClass, MyClassAssertions>
{
    public MyClassAssertions(MyClass subject) : base(subject) { }
    protected override string Identifier => nameof(MyClass);
}

public static class Extensions
{
    public static MyClassAssertions Should(this MyClass myClass) => new MyClassAssertions(myClass);
}
````
That allows:
````cs
myClass.Should()....
````

But how do you add a custom assertion for a type that already has assertions, e.g. `DateTime`:
````cs
myDateTime.Should().BeFooBarBaz();
````

**Comment by dennisdoomen** [comment, marked as answer]
Add an extension method that extends `DateTimeAssertions`. We do this in https://github.com/fluentassertions/fluentassertions/blob/fb53da18fff7c92d917bb20ba4d50f0271f2a08f/Src/FluentAssertions/NumericAssertionsExtensions.cs#L668
